src/dialects/postgres/query-generator.js

```diff
@@ -392,10 +394,39 @@ export class PostgresQueryGenerator extends AbstractQueryGenerator {
       primaryKeys = model.primaryKeyAttributes.length > 1 ? `(${pks})` : pks;
       primaryKeysSelection = pks;
 
-      return `DELETE FROM ${table} WHERE ${primaryKeys} IN (SELECT ${primaryKeysSelection} FROM ${table}${whereClause}${limit})`;
+      query = `${baseQuery} WHERE ${primaryKeys} IN (SELECT ${primaryKeysSelection} FROM ${table}${whereClause}${limit})`;
+      if (options.returning) {
+        returnValues = this.generateValuesToReturn(options);
+        query = `${query}${returnValues}`;
+      }
+
+      return query;
     }
 
-    return `DELETE FROM ${table}${whereClause}`;
+    if (options.returning) {
+      returnValues = this.generateValuesToReturn(options);
+      query = `${baseQuery}${whereClause}${returnValues}`;
+
+      return query;
+    }
+
+    return `${baseQuery}${whereClause}`;
+  }
+
+  generateValuesToReturn(options) {
```

Comment:
I somehow have the feeling that something like this might exist somewhere already.

Reply:
We do indeed have similar code in the abstract query generator for returning values for inserted/updates fields;
https://github.com/sequelize/sequelize/blob/2deee3c37ea3ff9e5d159538b3a1c067fc6d3ba2/src/dialects/abstract/index.ts#L26-L30
https://github.com/sequelize/sequelize/blob/2deee3c37ea3ff9e5d159538b3a1c067fc6d3ba2/src/dialects/abstract/query-generator.js#L1844-L1890

Reply:
I used the updateQuery(postgres) as a template to structure the code for this deleteQuery.

Reply:
Cool. Can we utilize the existing function?

Reply:
Updated code to utilize the existing one.